src/hotspot/cpu/ppc/assembler_ppc.hpp

```diff
@@ -1424,6 +1515,32 @@ class Assembler : public AbstractAssembler {
   static bool is_lis(int x) {
      return is_addis(x) && inv_ra_field(x)==0;
   }
+  static bool is_paddi_prefix(int x) {
+     return PADDI_PREFIX_OPCODE == (x & PADDI_PREFIX_OPCODE_MASK);
+  }
+  static bool is_paddi_suffix(int x) {
+     return PADDI_SUFFIX_OPCODE == (x & PADDI_SUFFIX_OPCODE_MASK);
+  }
+  static bool is_pli_prefix(int x) {
+     return is_paddi_prefix(x) && inv_r_eo(x) == 0;
+  }
+  static bool is_pli_suffix(int x) {
+     return is_paddi_suffix(x) && inv_ra_field(x) == 0;
+  }
+  static int is_paddi(const int* p, bool is_pli = false) {
+     int32_t* p_inst = (int32_t*)p;
+
+     if (is_aligned(reinterpret_cast<uintptr_t>(p_inst+1), 64) && is_nop(*p_inst)) {
```

Comment:
This test is a bit confusing.  Shouldn't is_paddi return false if p points at a nop (even if it precedes the paddi)?

Reply:
"is_<instr>()" are the functions for testing instructions in the generated code and they are usually called in "assert()" or complex combination of if statements.  So if this function returns false when p points at nop, the assertions or the if-conditions need to handle the nop case and the code will become difficult to read.  Since the padding nop is a non-common case, I'd like to hide the existence of nop in this function.

Reply:
Ok, got it.  Thanks for the explanation.  Maybe an extra comment in the code saying essentially what you said here would be appropriate?

Reply:
Looks good!